pkgs/games/fabric-servers/derivation.nix

```
@@ -0,0 +1,63 @@
+{
+  stdenvNoCC,
+  fetchurl,
+  makeWrapper,
+  lib,
+
+  jre,
+  udev,
+
+  version,
+  url,
+  hash,
+  ...
+}:
+
+with lib;
+
+stdenvNoCC.mkDerivation (final: {
+  pname = "fabric-server";
+  inherit version url hash;
+
+  src = fetchurl {
+    pname = "fabric-jar";
+    inherit version;
+
+    inherit (final) url hash;
+  };
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  dontUnpack = true;
+  dontBuild = true;
+
+  installPhase = ''
+    runHook preInstall
+
+    install -Dm644 $src $out/lib/fabric/server.jar
+
+    makeWrapper ${getExe jre} "$out/bin/minecraft-server" \
+      --append-flags "-jar $out/lib/fabric/server.jar nogui" \
+      ${optionalString stdenvNoCC.hostPlatform.isLinux "--prefix LD_LIBRARY_PATH : ${makeLibraryPath [ udev ]}"}
+
+    runHook postInstall
+  '';
+
+  preferLocalBuild = true;
+
+  passthru = {
+    updateScript = ./update.py;
+  };
+
+  meta = with lib; {
+    description = "Fabric is a modular, lightweight mod loader for Minecraft";
+    homepage = "https://fabricmc.net";
+    sourceProvenance = with sourceTypes; [ binaryBytecode ];
+    license = licenses.asl20;
+    platforms = platforms.unix;
+    maintainers = with maintainers; [
+      threadexio
+    ];
+    mainProgram = "minecraft-server";
+  };
```

```suggestion
  meta = {
    description = "Modular, lightweight mod loader for Minecraft";
    homepage = "https://fabricmc.net";
    sourceProvenance = with lib.sourceTypes; [ binaryBytecode ];
    license = lib.licenses.asl20;
    platforms = lib.platforms.unix;
    maintainers = with lib.maintainers; [
      threadexio
    ];
    mainProgram = "minecraft-server";
  };
```
About platforms: does the upstream support MacOS? (`platforms.unix` doesn't include it btw)